homeassistant/components/freedompro/climate.py

```diff
@@ -0,0 +1,144 @@
+"""Support for Freedompro climate."""
+import json
+import logging
+
+from pyfreedompro import put_state
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_HVAC_MODE,
+    HVAC_MODE_COOL,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_API_KEY, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+HVAC_MAP = {
+    0: HVAC_MODE_OFF,
+    1: HVAC_MODE_HEAT,
+    2: HVAC_MODE_COOL,
+}
+
+HVAC_INVERT_MAP = {
+    HVAC_MODE_OFF: 0,
+    HVAC_MODE_HEAT: 1,
+    HVAC_MODE_COOL: 2,
+}
+
+SUPPORTED_HVAC_MODES = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_COOL]
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro climate."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "thermostat"
+    )
+
+
+class Device(CoordinatorEntity, ClimateEntity):
+    """Representation of an Freedompro climate."""
+
+    def __init__(self, hass, api_key, device, coordinator):
```

Comment:
Please pass the session to the entity instead of `hass`.

Reply:
Sorry I didn't understand what you mean

Reply:
Instead of passing `hass` as a parameter here in the signature of the entity, pass the aiohttp client session. Ie get the client session first before creating the entity.